### DISCUSSION TITLE
Just curious about the relation between the deprecated webpack-hot-client and webpack-hot-middleware

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#18115] [Q&A] [answered]
As title. From the first few lines of [README.md](https://github.com/webpack-contrib/webpack-hot-client?tab=readme-ov-file#webpack-hot-client) of the deprecated webpack-hot-client it mentioned about webpack-dev-middleware:

> This is intended to work in concert with [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) and allows for adding Hot Module Replacement to an existing server, without a dependency upon [webpack-dev-server](https://github.com/webpack/webpack-dev-server). This comes in handy for testing in projects that already use server frameworks such as Express or Koa.

Here I'm interested in why it's deprecated and what's the alternative? What's the original problem it tried to solve and why it failed(since deprecated) to do so?

**Comment by alexander-akait** [comment, marked as answer]
It had bad design, just don't use it
